internal/transport/websocket_client.go

```diff
@@ -23,6 +24,8 @@ type WebsocketClient struct {
 	// Callback function called when the transport is closed
 	onClose func()
 
+	wg *sync.WaitGroup
```

Comment:
nit/ It is typical to store a value instead of a pointer, and then you can omit the explicit initialization in the constructor since the zero value is ready to use.
```suggestion
	wg sync.WaitGroup
```